### PR TITLE
Add no_gateway_ip option to os_subnet module

### LINKED ISSUE #1880: os_subnet does not support not having a Gateway IP
##### Issue Type:
- Bug Report
##### Ansible Version:

2.0.0
##### Ansible Configuration:

Valid clouds.yml
##### Environment:

N/A
##### Summary:

The [os_subnet module](https://github.com/ansible/ansible-modules-core/blob/devel/cloud/openstack/os_subnet.py) does not support creating / updating a neutron subnet without any gateway IP. This feature is necessary for creating a network that does not route to the outside world.
##### Steps To Reproduce:

```

---
- hosts: 127.0.0.1
  connection: local
  tasks:
    - name: create internal network
      os_network:
        name: "test-private"
        admin_state_up: yes
        shared: no
        cloud: "test"

    - name: add subnet to internal network
      os_subnet:
        name: "test-private-net"
        network_name: "test-private"
        cidr: "10.60.0.0/16"
        enable_dhcp: yes
        gateway_ip: null
        cloud: "test"
```
##### Expected Results:

I'd expect to be able to set some value for `gateway_ip` that results in the equivalent of `--no-gateway-ip`
in the [neutron CLI](http://docs.openstack.org/cli-reference/content/neutronclient_commands.html#neutronclient_subcommand_subnet-create). I check this by inspecting the created network in the Horizon UI and checking that the `Gateway IP` heading shows a dash `-`
##### Actual Results:

Currently, all I can do is leave out `gateway_ip` completely, which sets the gateway IP to the first valid IP in the subnet. I think that the underlying issue is that shade doesn't support `--no-gateway-ip`. I filed a bug for that issue.
https://github.com/openstack-infra/shade/blob/master/shade/__init__.py#L2719
https://storyboard.openstack.org/#!/story/2000328

### PR DESCRIPTION
**Pull request by saito-hideki**
no_gateway_ip option can provide subnet that is not having a gateway.
fixes #1880

### CODE REVIEW

**Comment by resmo** [comment]
Thanks @saito-hideki. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by saito-hideki** [comment]
Hi @emonty , I talked with you about this PR in OpenStack Summit Austin.  Would you please review it? Thanks.

**Comment by rcarrillocruz** [comment]
shipit

**Comment by saito-hideki** [comment]
Thanks!

File: cloud/openstack/os_subnet.py [hunk 1]
```diff
@@ -70,6 +70,11 @@
         - The ip that would be assigned to the gateway for this subnet
      required: false
      default: None
+   no_gateway_ip:
+     description:
+        - The gateway IP would not be assigned for this subnet
+     required: false
+     default: false
```
Comment (resmo) [inline cloud/openstack/os_subnet.py:78]: please add `version_added: "2.2"`

**Comment by gregdek** [comment]
Thanks again to @saito-hideki for this PR, and thanks @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/openstack/os_subnet.py (hunk 1 above)
Comment (saito-hideki) [inline cloud/openstack/os_subnet.py:78]: Done. Thanks!

**Comment by saito-hideki** [comment]
Hi @gregdek,

travis-ci was failed, but I don't know how to solve this issue...

**Comment by resmo** [comment]
@saito-hideki Thank you for this PR

I rebased and commited your work 215da19511b1402435c07de69c2379d375ad86cb. Somehow github removed the author of this commit :( Sorry for that, this was not my intention. 

Thanks again!

**Pull request closed without merging**